apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.ts

```diff
@@ -60,7 +139,94 @@ function getFilenames(locals) {
   return filenames;
 }
 
-async function pipeCursorToArchive(res, cursor: sqldb.CursorIterator<any>) {
+function parseFileContents(contents: any): Buffer | null {
+  if (contents == null) return null;
+
+  try {
+    return Buffer.from(typeof contents === 'string' ? contents : '', 'base64');
+  } catch {
+    // Ignore any errors in reading the contents and treat as a blank file.
+    return Buffer.from('');
+  }
+}
+
+interface RowForFiles {
+  submitted_answer: Submission['submitted_answer'];
+  params: Variant['params'];
+}
+
+function extractFiles<T extends RowForFiles>(
+  row: T,
+  makeFilename: (filename: string | null) => string | null,
+): ArchiveFile[] | null {
+  if (row.submitted_answer?.fileData) {
+    // Legacy v2 question data.
+    return [
+      {
+        filename: makeFilename(row.params?.fileName),
+        contents: parseFileContents(row.submitted_answer.fileData),
+      },
+    ];
+  }
+
+  // v3 question data.
+  return row.submitted_answer?._files?.map((file: any) => ({
+    filename: makeFilename(file.name),
+    contents: parseFileContents(file.contents),
+  }));
+}
+
+function extractFilesForSubmissions(row: AssessmentInstanceSubmissionRow): ArchiveFile[] | null {
+  // This doesn't handle QIDs with slashes in them:
+  // https://github.com/PrairieLearn/PrairieLearn/issues/7715
+  //
+  // We should probably rethink the directory structure that this will spit out.
+  const filenamePrefix = [
+    row.group_name ?? row.uid,
+    row.assessment_instance_number,
+    row.qid,
+    row.variant_number,
+    row.submission_number,
+    row.submission_id,
+  ].join('_');
+
+  return extractFiles(row, (suffix) => {
+    if (suffix == null) return null;
+
+    return filenamePrefix + '_' + suffix;
+  });
+}
+
+function extractFilesForManualGrading(row: ManualGradingSubmissionRow): ArchiveFile[] | null {
+  // This doesn't handle QIDs with slashes in them:
+  // https://github.com/PrairieLearn/PrairieLearn/issues/7715
+  //
+  // We should probably rethink the directory structure that this will spit out.
+  // We should also aim for more consistency between this function and
+  // `extractFilesForSubmissions`.
+  const filenamePrefix = [
+    row.group_name ?? [row.uid, row.uin].join('_'),
```

Comment:
https://github.com/PrairieLearn/PrairieLearn/pull/2614 removed `uin` from the "normal" file downloads, but not the manual grading ones. We should resolve this eventually, probably by removing it from the other file downloads as well.

It was originally added here: https://github.com/PrairieLearn/PrairieLearn/pull/2373